packages/vitest/src/node/test-run.ts

```diff
@@ -0,0 +1,53 @@
+import type { TaskResultPack } from '@vitest/runner'
+import type { ArgumentsType } from '../types/general'
+import type { Vitest } from './core'
+import type { Reporter } from './reporters'
+import type { TestModule } from './reporters/reported-tasks'
+import type { TestSpecification } from './spec'
+
+export class TestRun {
```

Comment:
I can start helping out with this. Did you already have an idea how this should be integrated into the `core.ts` and RPC calls?

Reply:
I wanted to have parsing logic here, and use higher level methods in `core` (`this.testRun.start(specs)`) and low lever in rpc calls (like `onTaskUpdate`)

This is also the place where we call reporter methods related to the test run.

The inspiration for method names is https://vshaxe.github.io/vscode-extern/vscode/TestRun.html

Reply:
I also think that proposed events are not enough, we will probably need to add more

Reply:
I also added `statistics` to the reported tasks, so we can modify them in place